plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/cwqTab/CwQTreeRootNode.kt

```diff
@@ -5,22 +5,72 @@ package software.aws.toolkits.jetbrains.core.explorer.cwqTab
 import com.intellij.ide.util.treeView.AbstractTreeNode
-import com.intellij.openapi.extensions.ExtensionPointName
+import com.intellij.openapi.actionSystem.ActionGroup
+import com.intellij.openapi.actionSystem.ActionManager
 import com.intellij.openapi.project.Project
-import software.aws.toolkits.jetbrains.core.explorer.cwqTab.nodes.CwQServiceNode
+import com.intellij.ui.SimpleTextAttributes
+import software.aws.toolkits.jetbrains.ToolkitPlaces
+import software.aws.toolkits.jetbrains.core.credentials.AwsBearerTokenConnection
+import software.aws.toolkits.jetbrains.core.credentials.ToolkitAuthManager
+import software.aws.toolkits.jetbrains.core.explorer.actions.AnActionTreeNode
+import software.aws.toolkits.jetbrains.core.gettingstarted.editor.ActiveConnection
+import software.aws.toolkits.jetbrains.core.gettingstarted.editor.BearerTokenFeatureSet
+import software.aws.toolkits.jetbrains.core.gettingstarted.editor.checkBearerConnectionValidity
+import software.aws.toolkits.jetbrains.services.amazonq.isQSupportedInThisVersion
+import software.aws.toolkits.jetbrains.services.codemodernizer.explorer.nodes.CodeModernizerRunModernizeNode
+import software.aws.toolkits.jetbrains.services.codemodernizer.isCodeModernizerAvailable
+import software.aws.toolkits.jetbrains.utils.isRunningOnRemoteBackend
+import software.aws.toolkits.resources.message
 
 class CwQTreeRootNode(private val nodeProject: Project) : AbstractTreeNode<Any>(nodeProject, Object()) {
+    private val runCodeModernizerNode by lazy { CodeModernizerRunModernizeNode(nodeProject) }
+
     override fun update(presentation: PresentationData) {}
 
-    override fun getChildren(): Collection<AbstractTreeNode<*>> = EP_NAME.extensionList
-        .filter { it.enabled() }
-        .map {
-            it.buildServiceRootNode(nodeProject).also { node ->
-                node.parent = this
+    override fun getChildren(): Collection<AbstractTreeNode<*>> {
+        val connection = checkBearerConnectionValidity(project, BearerTokenFeatureSet.Q)
+
+        val groupId = when (connection) {
+            is ActiveConnection.NotConnected -> if (otherActiveIdentityConnectionsAvailable()) ENABLE_Q_ACTION_GROUP else Q_SIGNED_OUT_ACTION_GROUP
+            is ActiveConnection.ValidBearer -> Q_SIGNED_IN_ACTION_GROUP
+            else -> Q_EXPIRED_TOKEN_ACTION_GROUP
+        }
+
+        val actions = ActionManager.getInstance().getAction(groupId) as ActionGroup
+        val childNodes = actions.getChildren(null).mapNotNull {
+            object : AnActionTreeNode(project, ToolkitPlaces.CWQ_TOOL_WINDOW, it) {
+                override fun update(presentation: PresentationData) {
+                    super.update(presentation)
+
+                    // We only want to display not supported info in the "Open Chat Panel" node not the migration node
+                    if (presentation.coloredText[0].text != message("action.q.openchat.text")) return
+
+                    if (groupId == Q_SIGNED_IN_ACTION_GROUP) {
```

Comment:
This if block is repeated for the Transform node too, so it could probably be declared in a common place. Not a blocker though